hadoop-hdds/managed-rocksdb/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedRocksObjectUtils.java

```diff
@@ -41,27 +42,24 @@ private ManagedRocksObjectUtils() {
 
   public static final Logger LOG =
       LoggerFactory.getLogger(ManagedRocksObjectUtils.class);
+
   private static final Duration POLL_DELAY_DURATION = Duration.ZERO;
   private static final Duration POLL_INTERVAL_DURATION = Duration.ofMillis(100);
 
-  public static void assertClosed(ManagedObject<?> object) {
-    assertClosed(object.get(), object.getStackTrace());
-  }
 
-  static void assertClosed(RocksObject rocksObject, String stackTrace) {
+  private static final LeakDetector LEAK_DETECTOR = new LeakDetector("ManagedRocksObject");
+
+  static LeakTracker track(AutoCloseable object, @Nullable StackTraceElement[] stackTrace) {
```

Comment:
- Use `RocksObject` instead of `AutoCloseable`.
- Get `stackTrace` inside this method instead of passing it in order to reduce code duplication.

Reply:
> * Get `stackTrace` inside this method instead of passing it in order to reduce code duplication.

This one I also included in HDDS-10000.  Since the `stackTrace` members already exist, i.e. the duplication is not introduced here, I thought it can be done as a follow-up.

Reply:
There's also `RocksMutableObject` and other custom object in the same package to be tracked here. `AutoCloseable ` is the most convenient here. Btw, it's package private and only used for managed object in this package.

Reply:
>Use RocksObject instead of AutoCloseable.
Get stackTrace inside this method instead of passing it in order to reduce code duplication.

This is great. Thanks.

Reply:
Increment `startIndex` (to 4) here:

https://github.com/apache/ozone/blob/333e6a1dcf1ab287d226225ce765c4988042957b/hadoop-hdds/managed-rocksdb/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedRocksObjectUtils.java#L73-L74